shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/adapter/ConnectionAdapterTest.java

```diff
@@ -172,6 +169,17 @@ private void assertClose(final ShardingSphereConnection actual) {
         assertTrue(cachedConnections.isEmpty());
     }
     
+    @Test
+    public void assertCloseShouldNotClearTransactionType() throws SQLException {
+        TransactionTypeHolder.set(TransactionType.XA);
+        TransactionType currentTransactionType = TransactionTypeHolder.get();
+        try (ShardingSphereConnection actual = getShardingSphereDataSource().getConnection()) {
+            actual.createStatement().executeQuery(sql);
+            actual.close();
+        }
+        assertEquals(currentTransactionType, TransactionTypeHolder.get());
```

Comment:
we usually use `assertThat(TransactionTypeHolder.get(), is(Transactiontype.XA))` to do assert. you can see more detail in our test case requirement from https://shardingsphere.apache.org/community/cn/contribute/code-conduct/

Reply:
done
I will read the specification carefully